coordinator/src/main/scala/com/socrata/datacoordinator/service/Service.scala

```diff
@@ -221,7 +224,32 @@ class Service(serviceConfig: ServiceConfig,
       "dataset" -> JString(formatDatasetId(name)),
       "schema" -> jsonifySchema(schema))
 
+  // Repeatedly tries to get a new thread by bumping numThreads, using
+  // compareAndSet to make sure our value is valid
+  @annotation.tailrec
+  private def tryGetMutationThread(numTries: Int = 100): Boolean = {
+    if (numTries <= 0) {
+      log.info("tryGetMutationThread: out of tries")
+      false
+    } else {
+      val currThreads = numThreads.get()
+      if (currThreads >= serviceConfig.maxMutationThreads) {
+        log.info(s"tryGetMutationThread: too many threads ($currThreads), waiting for some to finish")
+        Thread sleep 100
+        tryGetMutationThread(numTries - 1)
+      } else if (numThreads.compareAndSet(currThreads, currThreads + 1)) {
```

Comment:
Can we combine these two conditions with sleeps/recurstion into a single (currThreads >= serviceConfig.maxMutationThreads || !numThreads.compareAndSet(currThreads, currThreads + 1)?